src/app/[locale]/components/Navigation.tsx

```diff
@@ -0,0 +1,81 @@
+"use client";
+import Link from "next/link";
+import { useTasks } from "../context/useContext";
+import { useTranslations } from "next-intl";
+import { usePathname, useRouter } from "next/navigation";
+
+export default function Navigation({ locale }: { locale: string }) {
+  const { isLoggedIn, logout } = useTasks();
+  const isAuthenticated = isLoggedIn && isLoggedIn._id ? isLoggedIn : false;
+  const t = useTranslations("translations");
+  const pathname = usePathname();
+  const router = useRouter();
+
+  const changeLang = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const newLocale = e.target.value;
+    const path = pathname.split("/").slice(2).join("/");
+    router.push(`/${newLocale}/${path}`);
+  };
+
+  return (
+    <div className="container w-full mx-auto mt-2 p-4 flex justify-between drop-shadow bg-slate-100">
+      {isAuthenticated && isAuthenticated?.roles.includes("admin") && (
+        <Link
+          href={`/${locale}/createtask`}
```

Comment:
routes usually are written in snake case (create-task, task-list)

Reply:
also instead of checking each nav item with isAuthenticated, you can conditionally show the navbar if the user is logged in and then add these additional admin checks in this component

Reply:
fixed